federatedscope/vertical_fl/README.md

```diff
@@ -10,4 +11,170 @@ More details of the provided example can be found in [Tutorial](https://federate
 
-More support for vertical federated learning is coming soon! We really appreciate any contributions to FederatedScope !
\ No newline at end of file
+## Tree-based Models
+FederatedScope-Tree is built for training tree-based models in vertical federated learning, such as XGB, GBDT, RF, etc. 
+We provide a hands-on tutorial here. 
+
+We categorize tree-based models in vertical FL into two types, i.e., *feature-gathering tree-based models* and *label-scattering tree-based models*, according to their communication and computation protocols, and provide the corresponding implementations.
+- Feature-gathering tree-based models: The data parties (i.e., the participants who don't hold labels) send *the orders of their feature values* to the task party (i.e., the participant holds labels) for training, and then the task party calculate and compare the information gains for determining the split rules at the nodes. 
+- Label-scattering tree-based models: The task party sends the label-related information (such as gradient and hessian values) to the data parties for calculating the information gains, and all the gains would be retuned to the task parties for compression.
+
+
+### Configurations
+You can set the model and algorithm in yaml files as below:
+```bash
+use_gpu: False  # Whether to use GPU
+device: 0  # Deciding which GPU to use
+backend: torch
+
+# Federate learning related options
+federate:
+  mode: standalone  # `standalone` or `distributed`
+  client_num: 2  # number of client
+model:
+  type: xgb_tree  # xgb_tree or gbdt_tree or random_forest
+  # related hyperparameters
+  lambda_: 0.1
+  gamma: 0
+  num_of_trees: 10
+  max_tree_depth: 6
+  
+# Dataset related options
+data:
+  root: data/  # Root directory where the data stored
+  type: abalone  # Dataset name
+  splits: [0.8, 0.2]  # splits for training and testing
+dataloader:
+  type: raw  # Personalized DataLoader
+  batch_size: 4177
+criterion:
+  type: RegressionMSELoss  # CrossEntropyLoss, for binary classification
+
+# Trainer related options
+trainer:
+	# Trainer type
+  type: verticaltrainer  
+  
+# vertical related options
+vertical:
+  use: True
+  dims: [4, 8] # feature split for two clients, one has feature 0~3, 
+  # and the other has feature 4~7
+  feature_subsample_ratio: 1.0 # default = 1.0 
+  # the proportion of the numbers of features used for training per user.
+  algo: 'xgb' # 'xgb' or 'gbdt' or 'rf'
+  data_size_for_debug: 0  # use a subset for debug in vfl,
+  # 0 indicates using the entire dataset (disable debug mode)
+
+# Evaluation related options
+eval:
+	# Frequency of evaluation
+  freq: 3
+  best_res_update_round_wise_key: test_loss
+```
+Users can specify ```model.type``` and ```vertical.algo``` to use different models. For examples,:
+-  XGBoost: `model.type = xgb_tree`,and `vertical.algo = 'xgb'`
```

Comment:
please add a space between the comma and 'and' (also for the following two lines)